lib/train/file.rb

```diff
@@ -147,5 +129,85 @@ def mounted?
 
       !mounted.nil? && !mounted.stdout.nil? && !mounted.stdout.empty?
     end
+
+    def md5sum
+      return perform_ruby_checksum(method) if defined?(@use_ruby_checksum)
+
+      @md5_command ||= case @backend.os.family
+                       when 'darwin'
+                         # `-r` reverses output to match `md5sum` behavior
+                         'md5 -r'
+                       when 'solaris'
+                         'digest -a md5'
+                       else
+                         'md5sum'
+                       end
+
+      return perform_checksum(:md5) unless @backend.os.family == 'windows'
+
+      perform_checksum_windows(:md5)
```

Comment:
I think this can be a bit more simple, something like:
```
method = :md5
if @hash_ruby_fallback
  hash, err = perform_checksum_windows(method) if os.family == 'windows'
  hash, err  = perform_checksum_cmd(method) if os.family != 'windows'
  if err == nil
    return hash
  end
end
@hash_ruby_fallback = true
return perform_checksum_ruby(method)
```